src/components/Game/Timer/Timer.js

```diff
@@ -2,58 +2,58 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
 import classes from './Timer.scss';
-import { computeMinutes, computeSeconds, printTwoDigits } from '../../../helpers';
+import { formatTimeLeft } from '../../../helpers';
+import { DURATION, INTERVAL } from '../../../constants/constants';
+
 
 class Timer extends Component {
     state = {
-        // 60 sec. only for testing reasons. it's gonna be more :)
-        seconds: 60,
+        timeLeft: DURATION,
     };
 
+    start = new Date().getTime();
```

Comment:
:thinking: 
this value is the time when a game starts so I think it would make more sense to place this logic inside `startTimer` function
let's imagine you would like to display `message + let's play button` before the game starts… (for example: "Are you ready? The rules are: 1) … 2) … Good luck!") then you want to start the game after `let's play button` is clicked and not immediately; this is hypothetical case but it shows that there is a place where this line fits better (`startTimer`)

also, I'm not 100% sure about it but I think this value should rather be a part of `this.state`…

Reply:
`start` doesn't mutate during component's life so I think it doesn't have to be in the `state` (?).

I use `start` in `getTimeLeft` function which is called twice, also in `componentDidMount`, so once I place `start` in `startTimer` it's unreachable for `getTimeLeft` in `componentDidMount` (and once I call in `componentDidMount` `startTimer` before setting new state strange things happen to the timer 😕).

Reply:
> start doesn't mutate during component's life so I think it doesn't have to be in the state (?).

I see your point and this is why I'm not 100% sure
I think it can work if you have a value like this as class property
but it's more "correct", more react-way and more flexible (ability to start the game after user action) if it's kept in state

please be aware you can assign initial state in a constructor as well

okay, I played a bit with it and this is what I ended up:
```
    componentDidMount() {
        this.startTimer();
    }

    componentWillUnmount() {
        this.stopTimer();
    }

    getTimeLeft() {
        const { start } = this.state;
        return DURATION - (new Date().getTime() - start);
    }

    startTimer = () => {
        this.setState({
            start: new Date().getTime(),
        }, () => { // now this.state.start is set
            // we call updateTimeLeft right away
            this.updateTimeLeft();
            // and every INTERVAL (first execution of setInterval callback is *after* INTERVAL)
            this.interval = setInterval(this.updateTimeLeft, INTERVAL);
        });
    };

    updateTimeLeft = () => {
        const { timeIsOver } = this.props;
        this.setState({
            timeLeft: this.getTimeLeft(),
        }, () => {
            const { timeLeft } = this.state;
            if (timeLeft <= 0) {
                this.stopTimer();
                timeIsOver();
            }
        });
    }
```
wdyt?

Reply:
👍 very nice

Reply:
:blush: